### PR TITLE
Refactor profiling headers

### PR DESCRIPTION
**Pull request by yinan1048576**
The goal is to denoise `prof_externs.h`, so that prof internal functions are only tied to places defining / using them without being exposed externally.

### CODE REVIEW

**Comment by yinan1048576** [comment]
Moving some of the declarations to the `prof_inlines_*` files still expose them out though, so there's no effective difference.

File: include/jemalloc/internal/prof_internal_externs.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#ifndef JEMALLOC_INTERNAL_PROF_INTERNAL_EXTERNS_H
```
Comment (interwq) [inline include/jemalloc/internal/prof_internal_externs.h]: Maybe keep these still in prof_externs? It’s a bit strange to have an _internal_externs

Comment (yinan1048576) [reply include/jemalloc/internal/prof_internal_externs.h]: Got rid of `prof_internal_externs.h` by -
- Moved `prof_getpid()` and `prof_get_default_filename()` back to `prof_externs.h`. After all, they're quite general and may be externally needed.
- Moved the two `*_mutex_choose()` functions into `prof_data.c` and made them static. Also relocated the corresponding mutexes there but exposed them so that they can still be handled in `prof.c`.

**Comment by yinan1048576** [comment]
Revised according to feedback. Additionally exposed the mutex in `prof_log.c` so that it can be handled during forking (which was previously missing).

File: include/jemalloc/internal/prof_inlines_b.h [hunk 2]
```diff
@@ -5,6 +5,9 @@
 #include "jemalloc/internal/sz.h"
 #include "jemalloc/internal/thread_event.h"
 
+prof_tdata_t *prof_tdata_init(tsd_t *tsd);
+prof_tdata_t *prof_tdata_reinit(tsd_t *tsd, prof_tdata_t *tdata);
```
Comment (interwq) [inline include/jemalloc/internal/prof_inlines_b.h]: Why not keep them in prof_extern.h? I don’t see why moving them into _inlines.h is better.

File: include/jemalloc/internal/prof_inlines_a.h [hunk 3]
```diff
@@ -3,6 +3,9 @@
 
 #include "jemalloc/internal/mutex.h"
 
+bool prof_idump_accum_impl(tsdn_t *tsdn, uint64_t accumbytes);
+void prof_idump_rollback_impl(tsdn_t *tsdn, size_t usize);
```
Comment (interwq) [inline include/jemalloc/internal/prof_inlines_a.h]: Similarly here.

File: src/prof.c [hunk 4]
```diff
@@ -1147,6 +1119,7 @@ prof_prefork0(tsdn_t *tsdn) {
 		for (i = 0; i < PROF_NTDATA_LOCKS; i++) {
 			malloc_mutex_prefork(tsdn, &tdata_locks[i]);
 		}
+		malloc_mutex_prefork(tsdn, &log_mtx);
```
Comment (interwq) [inline src/prof.c]: Split into a separate commit? This looks to be bug fix.

File: include/jemalloc/internal/prof_inlines_b.h (hunk 2 above)
Comment (yinan1048576) [reply include/jemalloc/internal/prof_inlines_b.h]: As I mentioned in the earlier comment, there's no effective difference, in the sense that these functions are still exposed out. The only difference is for better understanding by first time readers: it appears clearer that these functions are only tied to these `prof_inlines_*` files rather than needed externally.

However, if it's not worthwhile to do so, I'm open to putting them back.

Comment (interwq) [reply include/jemalloc/internal/prof_inlines_b.h]: I worry about the order of the headers (e.g. externs are included first); also the inline functions may change. I don’t think there are other inline headers doing this, so let’s not create an exception here.

**Comment by yinan1048576** [comment]
Revised according to feedback.
